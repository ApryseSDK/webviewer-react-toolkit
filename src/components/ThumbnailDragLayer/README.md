This is a thumbnail drag preview which displays the number of pages that are
selected.
