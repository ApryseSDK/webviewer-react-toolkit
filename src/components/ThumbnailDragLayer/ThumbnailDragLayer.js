'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.ThumbnailDragLayer = void 0;
var tslib_1 = require('tslib');
var classnames_1 = tslib_1.__importDefault(require('classnames'));
var react_1 = tslib_1.__importStar(require('react'));
var icons_1 = require('../../icons');
var ThumbnailDragLayer = function(_a) {
  var _b = _a.numFiles,
    numFiles = _b === void 0 ? 1 : _b,
    className = _a.className,
    divProps = tslib_1.__rest(_a, ['numFiles', 'className']);
  numFiles = numFiles || 1;
  (0, react_1.useEffect)(
    function() {
      if (!Number.isInteger(numFiles))
        throw new RangeError('numFiles must be an integer');
      if (!Number.isFinite(numFiles))
        throw new RangeError('numFiles must not be infinite');
      if (numFiles <= 0)
        throw new RangeError('numFiles must be a positive integer');
    },
    [numFiles],
  );
  var thumbnailDragLayerClass = (0, classnames_1.default)(
    'ui__base ui__thumbnailDragLayer',
    className,
  );
  return react_1.default.createElement(
    'div',
    tslib_1.__assign({}, divProps, { className: thumbnailDragLayerClass }),
    react_1.default.createElement(
      'div',
      { className: 'ui__thumbnailDragLayer__wrapper' },
      numFiles === 1
        ? react_1.default.createElement(icons_1.SinglePage, {
            className: 'ui__thumbnailDragLayer__icon',
          })
        : react_1.default.createElement(icons_1.MultiPage, {
            className: 'ui__thumbnailDragLayer__icon',
          }),
      numFiles > 1
        ? react_1.default.createElement(
            'div',
            { className: 'ui__thumbnailDragLayer__numFiles' },
            react_1.default.createElement(
              'span',
              { className: 'ui__thumbnailDragLayer__numFiles__wrapper' },
              numFiles,
            ),
          )
        : undefined,
    ),
  );
};
exports.ThumbnailDragLayer = ThumbnailDragLayer;
//# sourceMappingURL=ThumbnailDragLayer.js.map
