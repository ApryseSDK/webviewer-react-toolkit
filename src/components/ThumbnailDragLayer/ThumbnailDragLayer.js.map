{"version":3,"file":"ThumbnailDragLayer.js","sourceRoot":"","sources":["ThumbnailDragLayer.tsx"],"names":[],"mappings":";;;;AAAA,kEAAoC;AACpC,qDAAyD;AACzD,qCAAoD;AAU7C,IAAM,kBAAkB,GAAG,UAAC,EAAiE;IAA/D,IAAA,gBAAY,EAAZ,QAAQ,mBAAG,CAAC,KAAA,EAAE,SAAS,eAAA,EAAK,QAAQ,sBAAtC,yBAAwC,CAAF;IACvE,QAAQ,GAAG,QAAQ,IAAI,CAAC,CAAC;IACzB,IAAA,iBAAS,EAAC;QACR,IAAI,CAAC,MAAM,CAAC,SAAS,CAAC,QAAQ,CAAC;YAAE,MAAM,IAAI,UAAU,CAAC,6BAA6B,CAAC,CAAC;QACrF,IAAI,CAAC,MAAM,CAAC,QAAQ,CAAC,QAAQ,CAAC;YAAE,MAAM,IAAI,UAAU,CAAC,+BAA+B,CAAC,CAAC;QACtF,IAAI,QAAQ,IAAI,CAAC;YAAE,MAAM,IAAI,UAAU,CAAC,qCAAqC,CAAC,CAAC;IACjF,CAAC,EAAE,CAAC,QAAQ,CAAC,CAAC,CAAC;IAEf,IAAM,uBAAuB,GAAG,IAAA,oBAAU,EAAC,iCAAiC,EAAE,SAAS,CAAC,CAAC;IAEzF,OAAO,CACL,0DAAS,QAAQ,IAAE,SAAS,EAAE,uBAAuB;QACnD,uCAAK,SAAS,EAAC,iCAAiC;YAC7C,QAAQ,KAAK,CAAC,CAAC,CAAC,CAAC,CAChB,8BAAC,kBAAU,IAAC,SAAS,EAAC,8BAA8B,GAAG,CACxD,CAAC,CAAC,CAAC,CACF,8BAAC,iBAAS,IAAC,SAAS,EAAC,8BAA8B,GAAG,CACvD;YACA,QAAQ,GAAG,CAAC,CAAC,CAAC,CAAC,CACd,uCAAK,SAAS,EAAC,kCAAkC;gBAC/C,wCAAM,SAAS,EAAC,2CAA2C,IAAE,QAAQ,CAAQ,CACzE,CACP,CAAC,CAAC,CAAC,SAAS,CACT,CACF,CACP,CAAC;AACJ,CAAC,CAAC;AA1BW,QAAA,kBAAkB,sBA0B7B"}