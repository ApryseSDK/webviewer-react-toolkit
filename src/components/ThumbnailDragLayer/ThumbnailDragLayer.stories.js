'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.Basic = void 0;
var tslib_1 = require('tslib');
var react_1 = tslib_1.__importDefault(require('react'));
var integer_1 = require('../../storybook-helpers/knobs/integer');
var ThumbnailDragLayer_1 = require('../ThumbnailDragLayer');
var README_md_1 = tslib_1.__importDefault(require('./README.md'));
exports.default = {
  title: 'Components/ThumbnailDragLayer',
  component: ThumbnailDragLayer_1.ThumbnailDragLayer,
  parameters: { readme: README_md_1.default },
};
var Basic = function() {
  return react_1.default.createElement(
    ThumbnailDragLayer_1.ThumbnailDragLayer,
    { numFiles: (0, integer_1.integer)('numFiles', 1) },
  );
};
exports.Basic = Basic;
//# sourceMappingURL=ThumbnailDragLayer.stories.js.map
