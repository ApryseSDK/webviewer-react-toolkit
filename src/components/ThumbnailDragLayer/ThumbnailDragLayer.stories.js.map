{"version":3,"file":"ThumbnailDragLayer.stories.js","sourceRoot":"","sources":["ThumbnailDragLayer.stories.tsx"],"names":[],"mappings":";;;;AAAA,wDAA0B;AAC1B,iEAAgE;AAChE,4DAA2D;AAC3D,kEAAiC;AAEjC,kBAAe,EAAE,KAAK,EAAE,+BAA+B,EAAE,SAAS,EAAE,uCAAkB,EAAE,UAAU,EAAE,EAAE,MAAM,qBAAA,EAAE,EAAE,CAAC;AAE1G,IAAM,KAAK,GAAG,cAAM,OAAA,8BAAC,uCAAkB,IAAC,QAAQ,EAAE,IAAA,iBAAO,EAAC,UAAU,EAAE,CAAC,CAAC,GAAI,EAAxD,CAAwD,CAAC;AAAvE,QAAA,KAAK,SAAkE"}