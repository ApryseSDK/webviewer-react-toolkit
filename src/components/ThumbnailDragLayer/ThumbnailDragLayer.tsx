import classnames from 'classnames';
import React, { FC, HTMLAttributes, useEffect } from 'react';
import single_page from '../../icons/single_page-24px.svg';
import multi_page from '../../icons/multi_page-24px.svg';

export interface ThumbnailDragLayerProps extends HTMLAttributes<HTMLDivElement> {
  /**
   * Must be a positive integer (1, 2, 3...).
   * @default 1
   */
  numFiles?: number;
}

export const ThumbnailDragLayer: FC<ThumbnailDragLayerProps> = ({ numFiles = 1, className, ...divProps }) => {
  useEffect(() => {
    if (!Number.isInteger(numFiles)) throw new RangeError('numFiles must be an integer');
    if (!Number.isFinite(numFiles)) throw new RangeError('numFiles must not be infinite');
    if (numFiles <= 0) throw new RangeError('numFiles must be a positive integer');
  }, [numFiles]);

  const thumbnailDragLayerClass = classnames('ui__base ui__thumbnailDragLayer', className);

  return (
    <div {...divProps} className={thumbnailDragLayerClass}>
      <div className="ui__thumbnailDragLayer__wrapper">
        <img className="ui__thumbnailDragLayer__icon" src={numFiles === 1 ? single_page : multi_page} alt="pages" />
        {numFiles > 1 ? (
          <div className="ui__thumbnailDragLayer__numFiles">
            <span className="ui__thumbnailDragLayer__numFiles__wrapper">{numFiles}</span>
          </div>
        ) : null}
      </div>
    </div>
  );
};
