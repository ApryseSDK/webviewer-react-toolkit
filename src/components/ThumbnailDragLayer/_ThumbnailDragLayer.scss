.ui__thumbnailDragLayer {
  width: $thumbnail-dimensions;
  height: $thumbnail-dimensions;
  @include flex-center;

  &__icon {
    height: $thumbnail-dimensions / 4;
    width: $thumbnail-dimensions / 4;
  }

  &__numFiles {
    @include absolute-fill;
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    font-size: $font-size-default;

    &__wrapper {
      margin: -($padding-small);
      padding: $padding-tiny;
      border-radius: $border-radius-large;
      background-color: $blue-gray-3;
    }
  }

  &__wrapper {
    position: relative;
  }
}
