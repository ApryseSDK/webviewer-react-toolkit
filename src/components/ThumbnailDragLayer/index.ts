export { ThumbnailDragLayer as default, ThumbnailDragLayerProps } from './ThumbnailDragLayer';
