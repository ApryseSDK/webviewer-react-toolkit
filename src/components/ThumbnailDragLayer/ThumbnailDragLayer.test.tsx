import { shallow } from 'enzyme';
import React from 'react';
import ThumbnailDragLayer from '../ThumbnailDragLayer';

describe('ThumbnailDragLayer component', () => {
  it('renders its contents', () => {
    const thumbnailDragLayer = shallow(<ThumbnailDragLayer />);
    expect(thumbnailDragLayer.find('.ui__thumbnailDragLayer').length).toEqual(1);
  });

  it('snapshot renders default thumbnailDragLayer', () => {
    const thumbnailDragLayer = shallow(<ThumbnailDragLayer />);
    expect(thumbnailDragLayer).toMatchSnapshot();
  });
});
