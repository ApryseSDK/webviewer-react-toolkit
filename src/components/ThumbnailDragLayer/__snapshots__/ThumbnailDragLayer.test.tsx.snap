// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThumbnailDragLayer component snapshot renders default thumbnailDragLayer 1`] = `
<div
  className="ui__base ui__thumbnailDragLayer"
>
  <div
    className="ui__thumbnailDragLayer__wrapper"
  >
    <img
      alt="pages"
      className="ui__thumbnailDragLayer__icon"
      src={[Function]}
    />
  </div>
</div>
`;
